### PR TITLE
Support Compartment.import

### PR DESCRIPTION
**Pull request by kriskowal**
This change introduces support for `import` and `importNow` on Compartments.
This is largely a refactor of `@agoric/make-importer` and subsumes the needed components into `ses` proper to avoid creating an interface with potential for shearing versions.

This is an early draft.

- [x] Address problems with embedding Babel with Rollup #268.
- [x] Subsume and expand tests from make-importer.
- Add support for module maps.
  - [x] string values
  - [x] module namespace values
  - [x] ~arbitrary other objects (construct a synthetic module instance)~
- [x] Add Compartment.module(spec):ModuleNamespace.
- [x] And inter-compartment linkage.
- [x] Add tests for inter-compartment linkage.
- [x] Compartment.import, Compartment.importNow, and eventually, dynamic import, must throw an error if the Compartment was constructed without an importHook and resolveHook.
- [x] Move ./src/node.js into ./test. It should only be used for tests and will not be public.
- [x] Change ModuleNamespace to a Proxy for an eventual ModuleNamespaceDescriptor. Create a Compartment method that returns a ModuleNamespace for any given module specifier. The Proxy should throw an error for any interaction before the module has been loaded and analyzed. Thereafter, all interactions should be delegated to the namespace object with properties for each module export (the existing module namespace implementation). There should be a WeakMap from each module namespace proxy to the corresponding compartment’s private fields, so that the Compartment constructor can produce the appropriate linkage when one of these namespaces is introduced to another Compartment through its moduleMap argument.

Follow up changes that need to be moved into separate tracking tickets:
- [ ] Copy node.js module into an agoric/ses-*-plugin package.
- [ ] Create an agoric/ses-*-plugin that provides a resolveHook and importHook for full-web Compartments where the moduleSpecifier is a fully qualified URL, the possibility of multiple supported protocol schemes, and possibly a same-origin security policy.
- [ ] Create agoric/ses-*-plugin packages that can assemble compartments from a package-lock.json, yarn’s lock file format, lava moat descriptor, or web importmap.
- [ ] Create a ses-test package or module that exports the utilities for generating an importHook from a static map and a compartment root URL.
- [ ] Add support for dynamic import and implement the importMetaHook. Be sure to write the full specification of importMetaHook on the proposal for TC39.

### CODE REVIEW

File: packages/ses/src/compartment-shim.js [hunk 1]
```diff
@@ -72,7 +101,45 @@ export class Compartment {
...
+            module,
+          )} in parent compartment`,
+        );
+      } else {
+        const alias = moduleAliases.get(module);
+        if (alias != null) {
+          // Modules from other components.
+          aliases.set(specifier, alias);
+        } else {
+          // TODO create and link a synthetic module instance from the given namespace object.
```
Comment (erights) [inline packages/ses/src/compartment-shim.js:126]: How convinced are you that we do eventually want to do this TODO? I am not clear how to think about it.

File: packages/ses/src/compartment-shim.js [hunk 2]
```diff
@@ -72,7 +101,45 @@ export class Compartment {
...
     privateFields.set(this, {
+      resolveHook,
+      importHook,
+      aliases,
+      moduleRecords,
+      deferredExports,
+      instances,
       globalTransforms,
       globalObject,
     });
```
Comment (erights) [inline packages/ses/src/compartment-shim.js:145]: Might as well rename the getter on line 148 below from `global` to `globalThis` in this PR. I don't see any good reason to separate that into another PR.

File: packages/ses/src/compartment-shim.js [hunk 3]
```diff
@@ -109,6 +176,51 @@ export class Compartment {
...
```
Comment (erights) [inline packages/ses/src/compartment-shim.js:176]: Line 154 talks about `x` as an option, but it is actually a mandatory positional parameter on like 156.

On line 156, why is it named `x`?

File: packages/ses/src/compartment-shim.js (hunk 1 above)
Comment (kriskowal) [reply packages/ses/src/compartment-shim.js:126]: I’m pretty sure we’ll need this to be able to inject initial capability-bearing modules like `fs`. The alternative is to brew these modules in an endowed compartment with a single source file that converts the properties of the endowment into exported names, which would be ceremony for ceremony’s sake.

File: packages/ses/src/compartment-shim.js (hunk 3 above)
Comment (kriskowal) [reply packages/ses/src/compartment-shim.js:176]: In this case, this is where I found Chesterton’s Fence. I would be elated to rename this `source` or `program` and fix the comment.

File: packages/ses/src/compartment-shim.js (hunk 2 above)
Comment (kriskowal) [reply packages/ses/src/compartment-shim.js:145]: This has turned out to be more invasive than I expected and I would like to capture this in a subsequent change.
